platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -804,15 +852,15 @@ def is_sw_control(self):
             return False
 
         db = utils.DbUtils.get_db_instance('STATE_DB')
-        control_type = db.get('STATE_DB', f'TRANSCEIVER_MODULES_MGMT|{self.sdk_index}', 'control_type')
-        control_file_value = utils.read_int_from_file(f'/sys/module/sx_core/asic0/module{self.sdk_index}/control')
-
-        if control_type == 'SW_CONTROL' and control_file_value == 1:
-            return True
-        elif control_type == 'FW_CONTROL' and control_file_value == 0:
-            return False
-        else:
-            raise Exception(f'Module {self.sdk_index} is in initialization, please retry later')
+        logical_port = NvidiaSFPCommon.get_logical_port_by_sfp_index(self.sdk_index)
+        if not logical_port:
+            raise Exception(f'Module {self.sdk_index} is not present or in initialization')
+        
+        initialized = db.exists('STATE_DB', f'TRANSCEIVER_STATUS|{logical_port}')
+        if not initialized:
+            raise Exception(f'Module {self.sdk_index} is not present or in initialization')
+        
+        return utils.read_int_from_file(f'/sys/module/sx_core/asic0/module{self.sdk_index}/control') == 1
```

Comment:
@Junchao-Mellanox i am not sure why not using TRANSCEIVER_INFO to distinguish between SW vs FW control

Reply:
Hi, we are using TRANSCEIVER_STATUS table now.